module/version_reconfigure_test.go

```diff
@@ -0,0 +1,235 @@
+package module_test
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"go.viam.com/test"
+	goutils "go.viam.com/utils"
+	"go.viam.com/utils/pexec"
+	"go.viam.com/utils/rpc"
+
+	"go.viam.com/rdk/components/generic"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/robot"
+	"go.viam.com/rdk/robot/client"
+	"go.viam.com/rdk/testutils"
+	"go.viam.com/rdk/utils"
+)
+
+// TODO(pre-merge) refactor this somewhat-shared function into a test_helpers esque file.
+func connect(port string, logger golog.Logger) (robot.Robot, error) {
+	connectCtx, cancelConn := context.WithTimeout(context.Background(), time.Second*30)
+	defer cancelConn()
+	for {
+		dialCtx, dialCancel := context.WithTimeout(context.Background(), time.Millisecond*500)
+		rc, err := client.New(dialCtx, "localhost:"+port, logger,
+			client.WithDialOptions(rpc.WithForceDirectGRPC()),
+			client.WithDisableSessions(), // TODO(PRODUCT-343): add session support to modules
+		)
+		dialCancel()
+		if !errors.Is(err, context.DeadlineExceeded) {
+			return rc, err
+		}
+		select {
+		case <-connectCtx.Done():
+			return nil, connectCtx.Err()
+		default:
+		}
+	}
+}
+
+// TODO(pre-merge) refactor this somewhat-shared function into a test_helpers esque file.
```

Comment:
Interesting take. I was recently complaining that viam-server should accept `port` on the command line which would also solve this.